### PR TITLE
Fix control node transform animation jitter with pivot offset

### LINKED ISSUE #28804: Control node (TextureRect) scale not smooth with pivot offset
**Godot version:**
3.1.1-stable

**OS/device including version:**
Windows 10 - 64bit

**Issue description:**
With a pivot offset on a Control node (TextureRect) other than the default (e.g. in the center of the rect), the scale transformation will wiggle the control around. This is most apparent, if one tries to slowly annimate the scale property.

**Problem source:**
https://github.com/godotengine/godot/blob/ecceabea7e74b56080c8deacc4323ac18b4b47c9/scene/gui/control.cpp#L453 
In the function _update_canvas_item_transform() there is a workaround to avoid flickering when moving the pivot offset. What ends up happening is that the pivot point gets rounded on every transform, resulting in a choppy animation. Regardless if pixel snap is enabled or not.

I compiled Godot without this workaround and the problem was fixed. I did not find a regression doing so, I was able to move the pivot around without flickering. 

**Steps to reproduce:**
- add TextureRect
- set pivot offset to the center
- scale the rect on x or y axis

**Minimal reproduction project:**
[ScalePivotBug.zip](https://github.com/godotengine/godot/files/3166020/ScalePivotBug.zip)

### PR DESCRIPTION
**Pull request by georgwacker**
The workaround for moving the pivot is not needed anymore, in fact it causes all transforms applied to control nodes to jitter while animating (if a pivot offset is set).

This can be observed via AnimationPlayer and Tween.

The fix is to remove the obsolete workaround that causes this bug.

Fixes #28804

### CODE REVIEW

**Comment by groud** [comment]
> The workaround for moving the pivot is not needed anymore

What makes you think that ? I don't think something changed since the time I implemented that. 
Have you tried within a hierarchy where the parent is rotated / scaled ?

I am ok with the change, but we should be 100% sure that it does not cause a regression.

**Comment by georgwacker** [comment]
In testing, I didn't observe any flickering or change with the pivot move (and hold) tool. The same goes for changing the pivot by sliding the value in the inspector. 

This was with two nested parents with both scale and rotation applied, as I just tested again. The pivot was changed on both the parents and the child without flickering.

As far as I can tell, the setting "snap_controls_to_pixels" is not actually used anywhere else, was it added with the workaround or is it defunct? The setting is turned on by default and thus the workaround gets applied, rounding every transform that is applied to the control node.
To clarify, this is not the "use_pixel_snap" setting in the editor toolbar.

Even if there is an edge-case I didn't account for, this workaround directly affects smooth game animations in contrast to an editor feature. I had to patch it out in my local Godot version, because it was so noticeable in two game projects.

**Comment by jgodfrey** [comment]
As noted here:

https://github.com/godotengine/godot/issues/28804

I'm noticing this same issue (jittery scaling via Animation or Tween) on a TextureButton object in the official 3.2 stable Windows build. It's unclear to me what the state of this bug or the mentioned fix is.

Should I still be seeing this in 3.2?

**Comment by georgwacker** [comment]
The fix wasn't merged, so the behaviour is still present in 3.2.

I consider this to be bug and not a 4.0 enhancement. If you rely on control node animations, then this is definitely unwanted behaviour.

The transform should not be rounded on each update. `snap_controls_to_pixels` is turned on by default, which means the pivot tool workaround affects every control node transform. 

In my testing, the workaround was not needed. If this fix causes a regression with the pivot tool we should find a better way to implement it, as the workaround seems a bit hacky. An editor tool for changing a parameter should not influence the transform on every update.

If the pivot tool does indeed need a flicker workaround in some cases, it also shouldn't be tied to an option like it is now (and I think the option `snap_controls_to_pixels` isn't used anywhere else, so it is just for this workaround). It should work correctly globally.

**Comment by groud** [comment]
Here is the original issue the it was supposed to fix: #17890
If removing the workaround does not bring the old incorrect behavior, it could be removed, otherwise we will have to find a better solution.

**Comment by georgwacker** [comment]
Thanks for linking the reference issue!

I went through the test cases in #17890 by using the test project and could not replicate the initial issues on current master 9591ca3fe.
I actually found another issue: with the workaround enabled and a scale bigger than 1, the pivot tool wobbles the control node around at multiples of 45 degrees (with left click hold and move).  It is working smoothly without the workaround.

I guess since August 2018 the control node now supports non-integer positions correctly and that was the root cause? It was mentioned in the original PR.

On a side note, it might actually be nice to have the pivot tool snap to the grid, but that should be implemented in edit_set_pivot?

To summarise:
- the workaround causes the animation wobble and the tool wobble with specific scale/rotation
- the initial issue in #17890 is no longer present in 9591ca3fe

**Comment by groud** [comment]
In that case I see no problem in removing this workaround then, the PR looks good to merge to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by georgwacker** [comment]
Great, thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.1.

**Comment by akien-mga** [comment]
Cherry-pick reverted due to the regression #36087.
It can be cherry-picked again together with a regression fix once it is found.

**Comment by volzhs** [comment]
@akien-mga it seems not reverted yet on 3.2

**Comment by akien-mga** [comment]
It's in my local branch.
